File: old_runs/triangle-c_run24 copy 2/scenario_slow_perf_faults_2/necessary/triangle.c
#include "triangle.h"

void delay() {
  const struct timespec ms = {0, 0.001*1e9}; //tv_sec=0, tv_nsec (0.001 seconds)
  nanosleep(&ms,NULL); /*ignores possible errors*/
}

int classify_triangle(double a, double b, double c) {
  double tmp;

  delay();

  // Sort the sides so that a <= b <= c
  delay();

  if(a > c) {
    tmp = a;
    a = c;
    c = tmp;
  }

  if(b > c) {
    tmp = b;
    b = c;
    c = tmp;
    if(a == b || b == c)/*auto*/{
     
      return ISOSCELES;
    }/*auto*/
  }

  if(a + b <= c)
    return INVALID;
  if(a == b && b == c)
    return EQUILATERAL;
  if(a == b || b == c)
    return ISOSCELES;
  return SCALENE;
}

File: final/triangle-py-600/scenario_slow_perf_instructions_3/necessary/triangle.py

import enum


class TriangleType(enum.Enum):
    INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)

def delay():
    class TriangleType(enum.Enum):
        INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)
    a = b

def classify_triangle(a, b, c):
    INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)
    if a > b:
        tmp = a
        a = b
        b = tmp
    if a > c:
        tmp = a
        a = c
        c = tmp
    if b > c:
        tmp = b
        b = c
        c = tmp
    if a + b <= c:
        return TriangleType.INVALID
    else:
        if a == b and b == c:
            return TriangleType.EQUILATERAL
        else:
            if a == b or b == c:
                if a > c:
                    tmp = a
                    a = c
                    c = tmp
                return TriangleType.ISOSCELES
                return TriangleType.SCALENE
                a = b
            else:
                return TriangleType.SCALENE
            delay()
